x-pack/plugin/security/qa/smoke-test-all-realms/src/javaRestTest/java/org/elasticsearch/xpack/security/authc/SecurityRealmSmokeTestCase.java

```diff
@@ -0,0 +1,141 @@
+        String token = basicAuthHeaderValue("admin_user", new SecureString("admin-password".toCharArray()));
+        return Settings.builder()
+            .put(ThreadContext.PREFIX + ".Authorization", token)
+            .put(CERTIFICATE_AUTHORITIES, httpCAPath)
+            .build();
+    }
+
+    @Override
+    protected Settings restClientSettings() {
+        String token = basicAuthHeaderValue("security_test_user", new SecureString("security-test-password".toCharArray()));
+        return Settings.builder()
+            .put(ThreadContext.PREFIX + ".Authorization", token)
+            .put(CERTIFICATE_AUTHORITIES, httpCAPath)
+            .build();
+    }
+
+    @Override
+    protected String getProtocol() {
+        // Because http.ssl.enabled = true
+        return "https";
+    }
+
+    protected Map<String, Object> authenticate(RequestOptions.Builder options) throws IOException {
+        final Request request = new Request("GET", "_security/_authenticate");
+        request.setOptions(options);
+        final Response response = client().performRequest(request);
+        return entityAsMap(response);
+    }
+
+    protected void assertUsername(Map<String, Object> authenticateResponse, String username) {
+        assertThat(authenticateResponse, hasEntry("username", username));
+    }
+
+    protected void assertRealm(Map<String, Object> authenticateResponse, String realmType, String realmName) {
+        assertThat(authenticateResponse, hasEntry(equalTo("authentication_realm"), instanceOf(Map.class)));
+        Map<?, ?> realmObj = (Map<?, ?>) authenticateResponse.get("authentication_realm");
+        assertThat(realmObj, hasEntry("type", realmType));
+        assertThat(realmObj, hasEntry("name", realmName));
+    }
+
+    protected void assertRoles(Map<String, Object> authenticateResponse, String... roles) {
+        assertThat(authenticateResponse, hasEntry(equalTo("roles"), instanceOf(List.class)));
+        String[] roleJson = ((List<?>) authenticateResponse.get("roles")).toArray(String[]::new);
+        assertThat("Server returned unexpected roles list [" + Strings.arrayToCommaDelimitedString(roleJson) + "]",
+            roleJson,
+            arrayContainingInAnyOrder(roles)
+        );
+    }
+
+    protected void createUser(String username, SecureString password, List<String> roles) throws IOException {
+        final RestHighLevelClient client = getHighLevelAdminClient();
+        client.security().putUser(PutUserRequest.withPassword(new User(username, roles), password.getChars(), true,
+            RefreshPolicy.WAIT_UNTIL), RequestOptions.DEFAULT);
+    }
+
+    protected void createRole(String name, Collection<String> clusterPrivileges) throws IOException {
+        final RestHighLevelClient client = getHighLevelAdminClient();
+        final Role role = Role.builder().name(name).clusterPrivileges(clusterPrivileges).build();
+        client.security().putRole(new PutRoleRequest(role, null), RequestOptions.DEFAULT);
```

Comment:
Nit: I'd also use `RefreshPolicy.WAIT_UNTIL` for `PutRoleRequest` even though it is not necessary for existing use cases. The performance gain with `null` is trivial and it feels more future proof to use `WAIT_UNTIL` or even `IMMEDIATE`.